sdk/lib/crt/except/amd64/ehandler.c

```diff
@@ -1,19 +1,143 @@
+        EndAddress = ScopeTable->ScopeRecord[i].EndAddress;
+
+        /* Skip this scope if we are not within the bounds */
+        if ((IpOffset < BeginAddress) || (IpOffset >= EndAddress))
+        {
+            continue;
+        }
+
+        /* Check if this is an unwind */
+        if (ExceptionRecord->ExceptionFlags & EXCEPTION_UNWIND)
+        {
+            /* Check if this is a target unwind */
+            if (ExceptionRecord->ExceptionFlags & EXCEPTION_TARGET_UNWIND)
+            {
+                /* Check if the target is within the scope itself */
+                if ((TargetIpOffset >= BeginAddress) &&
+                    (TargetIpOffset <  EndAddress))
+                {
+                    return ExceptionContinueSearch;
+                }
+            }
+
+            /* Check if this is a termination handler / finally function */
+            if (ScopeTable->ScopeRecord[i].JumpTarget == 0)
+            {
+                /* Call the handler */
+                Handler = ScopeTable->ScopeRecord[i].HandlerAddress;
+                TerminationHandler = (PTERMINATION_HANDLER)(ImageBase + Handler);
+                TerminationHandler(TRUE, EstablisherFrame);
+            }
+            else if (ScopeTable->ScopeRecord[i].JumpTarget == TargetIpOffset)
+            {
+                return ExceptionContinueSearch;
+            }
+        }
+        else
+        {
+            /* We are only unterested in exception handlers */
+            if (ScopeTable->ScopeRecord[i].JumpTarget == 0)
+            {
+                continue;
+            }
+
+            /* This is an exception filter, get the handler address */
+            Handler = ScopeTable->ScopeRecord[i].HandlerAddress;
+
+            /* Check for hardcoded EXCEPTION_EXECUTE_HANDLER */
+            if (Handler == EXCEPTION_EXECUTE_HANDLER)
+            {
+                /* This is our result */
+                FilterResult = EXCEPTION_EXECUTE_HANDLER;
+            }
+            else
+            {
+                /* Otherwise we need to call the handler */
+                ExceptionFilter = (PEXCEPTION_FILTER)(ImageBase + Handler);
+                FilterResult = ExceptionFilter(&ExceptionPointers, EstablisherFrame);
+            }
+
+            if (FilterResult == EXCEPTION_CONTINUE_EXECUTION)
+            {
+                return ExceptionContinueExecution;
+            }
+
+            if (FilterResult == EXCEPTION_EXECUTE_HANDLER)
+            {
+                JumpTarget = (ImageBase + ScopeTable->ScopeRecord[i].JumpTarget);
+
+                /* Unwind to the target address */
+                RtlUnwindEx(EstablisherFrame,
+                            (PVOID)JumpTarget,
+                            ExceptionRecord,
+                            UlongToPtr(ExceptionRecord->ExceptionCode),
+                            DispatcherContext->ContextRecord,
+                            DispatcherContext->HistoryTable);
+                __debugbreak();
```

Comment:
debug leftover?

Reply:
Unreachable code. I think we have a macro somewhere. UNREACHABLE() ?

Reply:
UNREACHABLE is only a compiler hint.